src/utils/setUserFromApiById.ts

```diff
@@ -0,0 +1,16 @@
+import { Dispatch } from 'react';
+import { getUser } from '../api';
+import { User } from '../types/User';
+
+export function setUserFromApiById(
+  userId: number,
+  setUser: React.Dispatch<React.SetStateAction<User | undefined>>,
+  setLoading: Dispatch<React.SetStateAction<boolean>>,
+) {
+  getUser(userId)
+    .then(setUser)
+    .catch(() => alert('User api is wrong!'))
```

Comment:
The error message in the `catch` block is not very descriptive. Consider providing a more informative error message or logging the error to help with debugging. For example, you could log the error object to the console.